drivers/performance_measurement/app/models/performance_measurement/equity_analysis/data.rb

```diff
@@ -184,10 +195,154 @@ def chart_height
     end
 
     def calculate_height(groups)
-      bars = BARS.count * (BAR_HEIGHT + PADDING)
-      total = bars / RATIO
+      bar_height = bars.count * (BAR_HEIGHT + PADDING)
+      total = bar_height / RATIO
       height = groups.count * total
       [height, MIN_HEIGHT].max
     end
+
+    private def apply_census_params(investigate_by:, year:)
+      results = geometries.map do |geo|
+        geo.population(internal_names: census_code(investigate_by), year: year)
+      end
+
+      results.each do |result|
+        if result.error
+          Rails.logger.error "population error: #{result.msg}. Sum won't be right!"
+          return nil
+        elsif result.year != year
+          Rails.logger.warn "Using #{result.year} instead of #{year}"
+        end
+      end
+
+      results.map(&:val).sum.round
+    end
+
+    private def race_census_code(code)
+      case code
+      when 'AmIndAKNative' then 'NATIVE_AMERICAN'
+      when 'Asian' then 'ASIAN'
+      when 'BlackAfAmerican' then 'BLACK'
+      when 'NativeHIPacific' then 'PACIFIC_ISLANDER'
+      when 'White' then 'WHITE'
+      when 'RaceNone' then 'OTHER_RACE'
+      when 'MultiRacial' then 'TWO_OR_MORE_RACES'
+      when 'HispanicLatinaeo' then 'HISPANIC'
+      when 'All' then 'ALL_PEOPLE'
+      end
```

Comment:
Will `MidEastNAfrican` never turn up? Or is it OK, that it falls through to `nil`?

Reply:
Correct, non-census races will just be zero.  At least for now, that's all we can do.